src/components/PupilViews/PupilReview/PupilReview.view.tsx

```diff
@@ -82,27 +82,35 @@ export const PupilViewsReview = (props: PupilViewsReviewProps) => {
     </OakLessonBottomNav>
   );
   const handleShareResultsClick = async () => {
+    setIsAttemptingShare(true);
     const attemptData = {
       lessonData: { slug: lessonSlug, title: lessonTitle },
       browseData: { subject: subject, yearDescription: yearDescription ?? "" },
       sectionResults: sectionResults,
     };
     try {
-      const attemptId = await logAttempt(attemptData, false);
+      const parsedAttemptData = attemptDataCamelCaseSchema.parse(attemptData);
+      const attemptId = await logAttempt(parsedAttemptData, false);
```

Comment:
The wait for this is very long and will be unexpected by the user. I wonder whether we rejig this for a fire and forget approach.